go/cmd/orchestrator-agent/main.go

```diff
@@ -33,11 +33,22 @@ var AppVersion string
 
 func acceptSignal() {
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, os.Kill, syscall.SIGHUP)
 
 	// Block until a signal is received.
-	sig := <-c
-	log.Fatalf("Got signal: %+v", sig)
+	signal.Notify(c, syscall.SIGHUP, syscall.SIGKILL, syscall.SIGTERM, syscall.SIGTERM)
+	go func() {
```

Comment:
This doesn't block. Can you please explain the reasoning for changing from blocking to non-blocking?

Reply:
it's a just backported your code from here ;)
https://github.com/github/orchestrator/blame/master/go/logic/orchestrator.go#L138
i can restore to blocking behavior

Reply:
yes please.

Reply:
ok. i return blocking behavior